eml/screens/lectures/CompSwipeScreen.jsx

```diff
@@ -43,11 +44,43 @@ const CompSwipeScreen = ({ route }) => {
 	const [combinedLecturesAndExercises, setCombinedLecturesAndExercises] = useState([]);
 	const swiperRef = useRef(null);
 	const [resetKey, setResetKey] = useState(0);
+	const [studentId, setStudentId] = useState('');
+	const [lastStudyDate, setLastStudyDate] = useState('1970-01-01T00:00:00.000Z');
+
+	/**
+	 * Handles student study streak update process.
+	 * Checks difference in days between lastStudyDate and today.
+	 * If difference is greater than 0 it updates: studyStreak and lastStudyDate 
+	 * both in database, local storage and this local state.
+	 */
+	async function handleStudyStreak() {
+		try {		
+			const today = new Date();
+			const dayDifference = differenceInDays(new Date(lastStudyDate), today);	
+			
+			// Update study streak if it has not already been updated today
+			if (dayDifference > 0) {
+				const statusCode = await updateStudyStreak(studentId);	// Database
+				
+				if (statusCode !== 200) 
+					throw new Error();
```

Comment:
Minor: Could add an appropriate error message here

Reply:
That gets printed in the catch block: 
![image](https://github.com/user-attachments/assets/c989518d-da3c-4dd4-9aca-99e043321f63)